com.hazelcast.internal.nio.ascii.HttpRestEndpointGroupsTest

_5.0_ (commit 44128991bbf0d5d5500de62c9941fd0983b275fc)

Failed on release branch test (Oracle JDK 8): http://jenkins.hazelcast.com/view/Release/job/Hazelcast-5.releaseBranch-nightly/1/testReport/com.hazelcast.internal.nio.ascii/HttpRestEndpointGroupsTest/

Following tests failed:
- `testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]`
- `testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]`
- `testOthersWhenGroupEnabled[restEndpointGroup:HOT_RESTART]`
- `testOthersWhenGroupDisabled[restEndpointGroup:HOT_RESTART]`

Stacktrace for `testOthersWhenGroupEnabled` (for `[restEndpointGroup:PERSISTENCE]`):
```
java.lang.AssertionError: Connection close was expected (from server side). TestUrl [restEndpointGroup=HOT_RESTART, method=POST, requestUri=/hazelcast/rest/management/cluster/forceStart, expectedSubstring=HTTP/1.1 400 Bad Request]
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.hazelcast.internal.nio.ascii.RestApiConfigTestBase.assertNoTextProtocolResponse(RestApiConfigTestBase.java:151)
	at com.hazelcast.internal.nio.ascii.HttpRestEndpointGroupsTest.testOthersWhenGroupEnabled(HttpRestEndpointGroupsTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Stacktrace for `testOthersWhenGroupDisabled` (for `[restEndpointGroup:PERSISTENCE]`):
```
java.lang.AssertionError: 
TestUrl [restEndpointGroup=HOT_RESTART, method=POST, requestUri=/hazelcast/rest/management/cluster/forceStart, expectedSubstring=HTTP/1.1 400 Bad Request]
Expected: a string containing "HTTP/1.1 400 Bad Request"
     but: was ""
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:964)
	at com.hazelcast.internal.nio.ascii.AbstractTextProtocolsTestBase$1.run(AbstractTextProtocolsTestBase.java:51)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1260)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1277)
	at com.hazelcast.internal.nio.ascii.RestApiConfigTestBase.assertTextProtocolResponse(RestApiConfigTestBase.java:135)
	at com.hazelcast.internal.nio.ascii.HttpRestEndpointGroupsTest.testOthersWhenGroupDisabled(HttpRestEndpointGroupsTest.java:82)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output for `testOthersWhenGroupEnabled` (for `[restEndpointGroup:PERSISTENCE]`):
```
11:23:44,834  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [TcpServerConnection] hz.cool_nightingale.IO.thread-in-0 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Connection[id=10, /127.0.0.1:5216->/127.0.0.1:49992, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: REST endpoint group is not enabled - PERSISTENCE. To enable it, please do one of the following:
- Change member config using JAVA API:  config.getNetworkConfig().getRestApiConfig().enableGroups(RestEndpointGroup.PERSISTENCE);
- Change XML/YAML configuration property: hazelcast.network.rest-api.endpoint-group PERSISTENCE with `enabled` set to true
- Add system property: -Dhz.network.rest-api.endpoint-groups.persistence.enabled=true
- Add environment variable property: HZ_NETWORK_RESTAPI_ENDPOINTGROUPS.PERSISTENCE.ENABLED=true
11:23:53,033  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [TcpServerConnection] hz.keen_nightingale.IO.thread-in-0 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Connection[id=10, /127.0.0.1:5222->/127.0.0.1:43382, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
11:23:53,232  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [LifecycleService] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] [127.0.0.1]:5222 is SHUTTING_DOWN
11:23:53,232  WARN |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Terminating forcefully...
11:23:53,232  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Shutting down connection manager...
11:23:53,233  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Shutting down node engine...
11:23:53,234  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [NodeExtension] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Destroying node NodeExtension.
11:23:53,234  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
11:23:53,234  INFO |testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]| - [LifecycleService] Thread-8527 - [127.0.0.1]:5222 [testOthersWhenGroupEnabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] [127.0.0.1]:5222 is SHUTDOWN
```

Standard output for `testOthersWhenGroupDisabled` (for `[restEndpointGroup:PERSISTENCE]`):
```
11:23:55,190  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [LifecycleService] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] [127.0.0.1]:5216 is SHUTTING_DOWN
11:23:55,190  WARN |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Terminating forcefully...
11:23:55,190  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [TextCommandServiceImpl] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Stopping text command service...
11:23:55,190  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Shutting down connection manager...
11:23:55,191  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Shutting down node engine...
11:23:55,192  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [NodeExtension] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Destroying node NodeExtension.
11:23:55,192  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [Node] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
11:23:55,192  INFO |testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]| - [LifecycleService] Thread-8526 - [127.0.0.1]:5216 [testOthersWhenGroupDisabled[restEndpointGroup:PERSISTENCE]] [5.0-SNAPSHOT] [127.0.0.1]:5216 is SHUTDOWN
```

--- a/hazelcast/src/test/java/com/hazelcast/internal/nio/ascii/HttpRestEndpointGroupsTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/nio/ascii/HttpRestEndpointGroupsTest.java
@@ -48,7 +48,7 @@ public class HttpRestEndpointGroupsTest extends RestApiConfigTestBase {
     public void testGroupEnabled() throws Exception {
         HazelcastInstance hz = factory.newHazelcastInstance(createConfigWithEnabledGroups(restEndpointGroup));
         for (TestUrl testUrl : TEST_URLS) {
-            if (restEndpointGroup == testUrl.restEndpointGroup) {
+            if (restEndpointGroup.getCode() == testUrl.restEndpointGroup.getCode()) {
                 assertTextProtocolResponse(hz, testUrl);
             }
         }
@@ -58,7 +58,7 @@ public class HttpRestEndpointGroupsTest extends RestApiConfigTestBase {
     public void testGroupDisabled() throws Exception {
         HazelcastInstance hz = factory.newHazelcastInstance(createConfigWithDisabledGroups(restEndpointGroup));
         for (TestUrl testUrl : TEST_URLS) {
-            if (restEndpointGroup == testUrl.restEndpointGroup) {
+            if (restEndpointGroup.getCode() == testUrl.restEndpointGroup.getCode()) {
                 assertNoTextProtocolResponse(hz, testUrl);
             }
         }
@@ -68,7 +68,7 @@ public class HttpRestEndpointGroupsTest extends RestApiConfigTestBase {
     public void testOthersWhenGroupEnabled() throws Exception {
         HazelcastInstance hz = factory.newHazelcastInstance(createConfigWithEnabledGroups(restEndpointGroup));
         for (TestUrl testUrl : TEST_URLS) {
-            if (restEndpointGroup != testUrl.restEndpointGroup) {
+            if (restEndpointGroup.getCode() != testUrl.restEndpointGroup.getCode()) {
                 assertNoTextProtocolResponse(hz, testUrl);
             }
         }
@@ -78,7 +78,7 @@ public class HttpRestEndpointGroupsTest extends RestApiConfigTestBase {
     public void testOthersWhenGroupDisabled() throws Exception {
         HazelcastInstance hz = factory.newHazelcastInstance(createConfigWithDisabledGroups(restEndpointGroup));
         for (TestUrl testUrl : TEST_URLS) {
-            if (restEndpointGroup != testUrl.restEndpointGroup) {
+            if (restEndpointGroup.getCode() != testUrl.restEndpointGroup.getCode()) {
                 assertTextProtocolResponse(hz, testUrl);
             }
         }

